community/server/src/main/java/org/neo4j/server/rrd/RrdFactory.java

```diff
@@ -114,52 +121,16 @@ public RrdFactory( Config config, Logging logging )
         return result.toArray( new Sampleable[result.size()] );
     }
 
-    private String getDefaultRrdFile( GraphDatabaseAPI db ) throws IOException
-    {
-        return isEphemereal( db ) ? tempRrdFile() : new File( db.getStoreDir(), "rrd" ).getAbsolutePath();
-    }
-
-    protected String tempRrdFile() throws IOException
-    {
-        final File tempFile = File.createTempFile( "neo4j", "rrd" );
-        tempFile.delete();
-        tempFile.mkdir();
-
-        Runtime.getRuntime().addShutdownHook( new Thread()
-        {
-            @Override
-            public void run()
-            {
-                try
-                {
-                    FileUtils.deleteRecursively(tempFile);
-                }
-                catch ( IOException e )
-                {
-                    // Ignore
-                }
-            }
-        });
-
-        return tempFile.getAbsolutePath();
-    }
-
-    private boolean isEphemereal( GraphDatabaseAPI db )
+    private File getDefaultRrdFile( Config config ) throws IOException
     {
-        Config config = db.getDependencyResolver().resolveDependency( Config.class );
-
-        if ( config == null )
-        {
-            return false;
-        }
-        else
-        {
-            Boolean ephemeral = config.get( InternalAbstractGraphDatabase.Configuration.ephemeral );
-            return ephemeral != null && ephemeral;
-        }
+        String path = config.get( ServerInternalSettings.rrd_store ).getAbsolutePath();
+        //noinspection ResultOfMethodCallIgnored
+        File rrdStore = new File( path );
+        rrdStore.mkdirs();
```

Comment:
we need to check if directory was actually created (as example we do not have enough of karma to create new dir)